contrib/native/client/src/clientlib/drillClientImpl.cpp

```diff
@@ -595,6 +611,12 @@ connectionStatus_t DrillClientImpl::validateHandshake(DrillUserProperties* prope
 
     switch(this->m_handshakeStatus) {
         case exec::user::SUCCESS:
+            // Check if client needs auth/encryption and server is not requiring it
+            if(clientNeedsAuthentication(properties) || clientNeedsEncryption(properties)) {
```

Comment:
Generally, all error messages come from errmsgs.cpp so we can localize them when we need to.

Reply:
Also, do you need two different messages : auth required and encr required ?

Reply:
- Externalized the messages to errmsgs.cpp 
- Changed the error message to "Client needs a secure connection but server does not support... " to account for the case where auth and/or enc is required by the client but missing on the server